TLM/TLM/Custom/AI/CustomTransportLineAI.cs

```diff
@@ -138,52 +195,67 @@ public class CustomTransportLineAI : TransportLineAI { // TODO inherit from NetA
-#if DEBUG
-                if (debug)
-                    Log._Debug($"TransportLineAI.CustomStartPathFind({segmentID}, ..., {netService}, {netService2}, {vehicleType}, {skipQueue}): Started calculating path {path} for extVehicleType={extVehicleType}, startPosA=[seg={startPosA.m_segment}, lane={startPosA.m_lane}, off={startPosA.m_offset}], startPosB=[seg={startPosB.m_segment}, lane={startPosB.m_lane}, off={startPosB.m_offset}], endPosA=[seg={endPosA.m_segment}, lane={endPosA.m_lane}, off={endPosA.m_offset}], endPosB=[seg={endPosB.m_segment}, lane={endPosB.m_lane}, off={endPosB.m_offset}]");
-#endif
+                Log._DebugIf(
+                    logPathfind,
+                    () => $"TransportLineAI.CustomStartPathFind({segmentId}, ..., {netService}, " +
+                    $"{netService2}, {vehicleType}, {skipQueue}): Started calculating " +
+                    $"path {path} for extVehicleType={extVehicleType}, " +
+                    $"startPosA=[seg={startPosA.m_segment}, lane={startPosA.m_lane}, " +
+                    $"off={startPosA.m_offset}], startPosB=[seg={startPosB.m_segment}, " +
+                    $"lane={startPosB.m_lane}, off={startPosB.m_offset}], " +
+                    $"endPosA=[seg={endPosA.m_segment}, lane={endPosA.m_lane}, " +
+                    $"off={endPosA.m_offset}], endPosB=[seg={endPosB.m_segment}, " +
+                    $"lane={endPosB.m_lane}, off={endPosB.m_offset}]");
                 return false;
             }
 
-            CustomTransportLineAI.CheckSegmentProblems(segmentID, ref data);
+            CheckSegmentProblems(segmentId, ref data);
             return true;
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
+        [UsedImplicitly]
         private static bool GetStopLane(ref PathUnit.Position pos, VehicleInfo.VehicleType vehicleType) {
-            Log.Error($"CustomTransportLineAI.GetStopLane called.");
+            Log._DebugOnlyError("CustomTransportLineAI.GetStopLane called.");
             return false;
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void CheckSegmentProblems(ushort segmentID, ref NetSegment data) {
-            Log.Error($"CustomTransportLineAI.CheckSegmentProblems called.");
+        [UsedImplicitly]
+        private static void CheckSegmentProblems(ushort segmentId, ref NetSegment data) {
+            Log._DebugOnlyError($"CustomTransportLineAI.CheckSegmentProblems called.");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void CheckNodeProblems(ushort nodeID, ref NetNode data) {
-            Log.Error($"CustomTransportLineAI.CheckNodeProblems called.");
+        [UsedImplicitly]
+        private static void CheckNodeProblems(ushort nodeId, ref NetNode data) {
+            Log._DebugOnlyError($"CustomTransportLineAI.CheckNodeProblems called.");
```

Comment:
need to call Error, not _DebugOnlyError

Reply:
in Release the function is not eliminated, the Log call is eliminated